python_training/mimic.py

```diff
@@ -0,0 +1,186 @@
+
+    def __str__(self)-> str:
+        """
+        Returns a textual representation of the list.
+
+        :return: A string representing the list.
+        """
+        base_string = '[ '
+        for member in self:
+            base_string += '{}, '.format(member.value) if member is not self.last_member else '{}'.format(member.value)
+        base_string += ' ]'
+        return base_string
+
+    def __add__(self, other):
+        """
+        Adds one MyList object to another and returns a new MyList object comprised of the two.
+        :param other: The second MyList object we want to add to the current one.
+        :return: A new MyList object containing the values of both the lists.
+        """
+        if type(other) is not type(self):
+            raise TypeError("Can only concatenate MyList  object to another MyList object")
+        copy_of_first_list = copy.deepcopy(self)
+        copy_of_second_list = copy.deepcopy(other)
+        copy_of_first_list.last_member.next_member = copy_of_second_list.first_member
+        copy_of_first_list.last_member = copy_of_second_list.last_member
+        return copy_of_first_list
+
+    def __getitem__(self, key):
+        """
+        Allows accessing MyList members via []
+        :param key: The position of the member that should be returned.
+        :return: The ListMember that matches the given position.
+        """
+        if type(key) is not int:
+            raise TypeError
+        if key < 0 or key > self.list_length - 1:
+            raise KeyError
+        else:
+            for index, member in enumerate(self):
+                if index == key:
+                    return member.value
+
+
+class ListIterator:
+    """
+    A Iterator for MyList objects.
+    """
+    def __init__(self, list_object):
+        """
+        Initiates the ListIterator.
+        :param list_object:
+        """
+        self.list = list_object
+        self.current_iteration_object = None
+
+    def __iter__(self):
+        return self
+
+    def __next__(self):
+        """
+        Gets the next object in the list.
+        :return: The next object in the list.
+        """
+        if self.current_iteration_object is None:
+            self.current_iteration_object = self.list.first_member
+        else:
+            self.current_iteration_object = self.current_iteration_object.next_member
+
+        if self.current_iteration_object is not None:
+            return self.current_iteration_object
+        else:
+            raise StopIteration
+
+
+class ListMember:
+    """
+    A member of a MyList series.
+    """
+    def __init__(self, value):
+        """
+        Initiates the ListMember and sets the value it holds.
+        :param value: The vale that should be held in the member.
+        """
+        self.value = value
+        self.next_member = None
+
+
+def main():
+    my_list = MyList()
+    my_list.append(1)
+    my_list.append(2)
+    my_list.append(3)
+    print(my_list)
```

Comment:
For each print, add a comment (above) with the expected output.
(Easier for me to check :) )